politeiad/api/v2/v2.go

```diff
@@ -0,0 +1,512 @@
+}
+
+// RecordEditMetadata edits the metadata of a record.
+//
+// MDAppend appends metadata to a metadata stream. MDOverwrite overwrites a
+// metadata stream. If the metadata stream does not exist yet for either of
+// these arguments, a new metadata stream will be created.
+type RecordEditMetadata struct {
+	Challenge   string           `json:"challenge"` // Random challenge
+	Token       string           `json:"token"`     // Censorship token
+	MDAppend    []MetadataStream `json:"mdappend"`
+	MDOverwrite []MetadataStream `json:"mdoverwrite"`
+}
+
+// RecordEditMetadataReply is the reply to the RecordEditMetadata command.
+type RecordEditMetadataReply struct {
+	Response string `json:"response"` // Challenge response
+	Record   Record `json:"record"`
+}
+
+// RecordSetStatus sets the status of a record.
+//
+// MDAppend appends metadata to a metadata stream. MDOverwrite overwrites a
+// metadata stream. If the metadata stream does not exist yet for either of
+// these arguments, a new metadata stream will be created.
+type RecordSetStatus struct {
+	Challenge   string           `json:"challenge"` // Random challenge
+	Token       string           `json:"token"`     // Censorship token
+	Status      RecordStatusT    `json:"status"`
+	MDAppend    []MetadataStream `json:"mdappend"`
+	MDOverwrite []MetadataStream `json:"mdoverwrite"`
+}
+
+// RecordSetStatusReply is the reply to the RecordSetStatus command.
+type RecordSetStatusReply struct {
+	Response string `json:"response"` // Challenge response
+	Record   Record `json:"record"`
+}
+
+// RecordGet retrieves a record. If no version is provided the most recent
+// version will be returned.
+type RecordGet struct {
+	Challenge string `json:"challenge"`         // Random challenge
+	Token     string `json:"token"`             // Censorship token
+	Version   uint32 `json:"version,omitempty"` // Record version
+}
+
+// RecordGetReply is the reply to the RecordGet command.
+type RecordGetReply struct {
+	Response string `json:"response"` // Challenge response
+	Record   Record `json:"record"`
+}
+
+// RecordRequest is used to request a record. It gives the caller granular
+// control over what is returned. The only required field is the token. All
+// other fields are optional. All record files are returned by default unless
+// one of the file arguments is provided.
+//
+// Version is used to request a specific version of a record. If no version is
+// provided then the most recent version of the record will be returned.
+//
+// Filenames can be used to request specific files. If filenames is not W
+// then the specified files will be the only files that are returned.
+//
+// OmitAllFiles can be used to retrieve a record without any of the record
+// files. This supersedes the filenames argument.
+type RecordRequest struct {
+	Token        string   `json:"token"`
+	Version      uint32   `json:"version"`
+	Filenames    []string `json:"filenames"`
+	OmitAllFiles bool     `json:"omitallfiles"`
```

Comment:
Nit but this could be "omitempty"

Reply:
Fixed.  I also went through all new APIs (politeiad, politeiawww, plugins) and verified that all optional fields are `omitempty`.

https://github.com/lukebp/politeia/blob/tlog/politeiad/api/v2/v2.go#L378